vsintegration/src/vs/FsPkgs/FSharp.Project/Common.Source.CSharp/Project/ProjectConfig.cs

```diff
@@ -1625,11 +1655,60 @@ internal bool IsUpToDate(OutputWindowLogger logger, bool testing)
                     freshestInputTime = timeStamp.Value;
             }
 
+            // check 1-1 mappings
+            foreach (var kv in directMappings)
+            {
+                string input = kv.Item1;
+                string copyToOutputDirectory = kv.Item2;
+                string output = kv.Item3;
+
+                var inputTime = TryGetLastWriteTimeUtc(input, logger);
+                if (!inputTime.HasValue)
+                {
+                    logger.WriteLine("Declaring project NOT up to date, can't find expected input {0}", input);
+                    return false;
+                }
+
+                if (String.Compare(copyToOutputDirectory, "Never", StringComparison.OrdinalIgnoreCase) == 0)
+                {
+                    //No need to check output
```

Comment:
This is not valid - just because a file is not copied somewhere as part of build does not mean it doesn't contribute toward the build.  Even inputs that don't map to specific outputs should still be considered as potentially causing the project to be out-of-date.